### PR TITLE
[DEVOPS-810] reduce the useless rebuilds in wallet-new due to nix and gitrev

### PR DESCRIPTION
**Pull request by cleverca22**
[DEVOPS-810]

todo:
- [ ] cardano-sl-auxx
- [ ] cardano-sl-explorer-static
- [ ] cardano-sl-tools
- [ ] cardano-sl-node
- [ ] cardano-sl-node-static

### CODE REVIEW

File: wallet-new/cardano-sl-wallet-new.cabal [hunk 1]
```diff
@@ -14,7 +14,10 @@ extra-source-files:  README.md
 cabal-version:       >=1.10
 
 library
-  hs-source-dirs:      src
+  hs-source-dirs:      src server
```
Comment (parsonsmatt) [inline wallet-new/cardano-sl-wallet-new.cabal]: If we're going to do this I want to move all the `server` code into `src`. Can happen at a later date/PR.

File: wallet-new/cardano-sl-wallet-new.cabal [hunk 2]
```diff
@@ -14,7 +14,10 @@ extra-source-files:  README.md
...
+  ghc-options:         -Wall
+                       -fno-warn-orphans
```
Comment (parsonsmatt) [inline wallet-new/cardano-sl-wallet-new.cabal]: :(

I'd rather put this in each file than here. Also can be another PR.

Comment (cleverca22) [reply wallet-new/cardano-sl-wallet-new.cabal]: i believe that was in the old cabal file, just moved a bit
